w06h01/test/pgdp/LoopTest.java

```diff
@@ -0,0 +1,146 @@
+package pgdp;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Stack;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+public class LoopTest {
+    // Note: This test has a lot of false positives (e.g. using the word "for"
+    // in a method name or a comment). The test should just warn you if you use
+    // loops in your solution.
+    @DisplayName("Source code should not contain loops because they are not allowed in this exercise")
+    @Test
+    public void testUsageOfLoops() {
+        String[] filePaths = new String[] {
+                "src/pgdp/datastructures/lists/RecIntList.java",
+                "src/pgdp/datastructures/lists/RecIntListElement.java"
+        };
+
+        String[] notAllowedKeywords = new String[] { "for", "while", "Stream" };
+
+        for (String path : filePaths) {
+            // Methods that are allowed to use loops
+            String[] allowedMethods = new String[] {
+                    "public static void main(String[] args) {",
+                    "public String toString() {",
+                    "public String toConnectionString() {"
+            };
+
+            String file = readFile(path);
+            String filteredFile = removeMultipleMethods(file, allowedMethods);
+
+            // Check if there are any loops
+            for (String keyword : notAllowedKeywords) {
+                assertTrue(!filteredFile.contains(keyword),
```

Comment:
This could lead to unexpected behavior, for example when you declare a variable with a name containing "for". Instead we should use a regular expression to match those keywords exactly. The regular expression could look something like this: /for\s*[(]{1}/gm (It checks if it is followed by an arbitrary amount of spaces and an opening bracket). This regex is not entirely bulletproof, but it should be enough for our case.